core/src/main/scala/kafka/log/Log.scala

```diff
@@ -718,7 +718,7 @@ class Log(@volatile var dir: File,
               records.records.asScala.map(_.offset))
           if (appendInfo.firstOrLastOffset < nextOffsetMetadata.messageOffset) {
             val firstOrLast = if (appendInfo.firstOffset.isDefined) "First" else "Last"
-            throw new IllegalArgumentException(s"Out of order offsets found in append to $topicPartition. $firstOrLast " +
+            throw new IllegalArgumentException(s"Unexpected offset in append to $topicPartition. $firstOrLast " +
               s"offset ${appendInfo.firstOrLastOffset} is less than the next offset ${nextOffsetMetadata.messageOffset}. " +
               s"Offsets in append: ${records.records.asScala.map(_.offset)}")
```

Comment:
Do you think it's still useful to print the individual offsets in this case? This message can get quite large in practice.

Reply:
I did wonder. I thought this was rare enough that it was better to have the information than not. But if we think the large message could be an issue, we could remove it or take the first n.

Reply:
The first n seems reasonable. I think in 100% of the cases that I have seen this error, the problem was the first offset.

Reply:
I'd vote for n === 1 :)